browser/extensions/brave_component_loader.cc

```diff
@@ -153,4 +161,49 @@ void BraveComponentLoader::AddWebTorrentExtension() {
   }
 }
 
+bool BraveComponentLoader::UseBraveExtensionBackgroundPage() {
+  // Keep sync with `pref_change_registrar_` in the ctor.
+  return profile_prefs_->GetBoolean(brave_rewards::prefs::kEnabled) ||
+         profile_prefs_->GetBoolean(kWebDiscoveryEnabled);
+}
+
+void BraveComponentLoader::UpdateBraveExtension() {
+  const base::CommandLine& command_line =
+      *base::CommandLine::ForCurrentProcess();
+  if (command_line.HasSwitch(switches::kDisableBraveExtension)) {
+    return;
+  }
+
+  base::FilePath brave_extension_path(FILE_PATH_LITERAL(""));
+  brave_extension_path =
+      brave_extension_path.Append(FILE_PATH_LITERAL("brave_extension"));
+  auto& resource_bundle = ui::ResourceBundle::GetSharedInstance();
+  std::optional<base::Value::Dict> manifest = base::JSONReader::ReadDict(
+      resource_bundle.GetRawDataResource(IDR_BRAVE_EXTENSION));
+  CHECK(manifest) << "invalid Brave Extension manifest";
```

Comment:
There's something going on with this check when building our browser tests in `Debug` mode, where this value returned by the resource bundle call will always be empty. I had seen failures before in Linux, a few weeks ago in fact, but it hadn't occurred to me that those were specific to debug builds.

```
npm run build -- Debug --ignore_compile_failure --target=brave_browser_tests 
../out/Debug/brave_browser_tests --test-launcher-jobs=1 --gtest_filter=BraveContentBrowserClientTest.CanLoadChromeURL
```

@atuchin-m Would you mind taking a look into this?

Reply:
https://github.com/brave/brave-browser/issues/41604

Reply:
I will take a look. The current hypothesis is that the resource bundle isn't property initialized in Debug browser tests.